SU-HSDO/suhumsci
# STN-185 : Search Results

# [STN-185](https://sparkbox.atlassian.net/browse/STN-185) READY FOR CODE REVIEW

## Summary
Implement the search results page as a stack of horizontal cards. Fill in the styling and pager functionality gaps as defined in [STN-185](https://sparkbox.atlassian.net/browse/STN-185).

The work in this PR has been reviewed by a designer (Merani).

## Need Review By (Date)
4/8/2020

## Urgency
low

## Steps to Test
1. In the CLI, run `npm run test`. Verify all tests pass.
    * _Note:_ As of the writing of this PR there was one failing test. Work to fix the failing test is being completed in another PR.
2. To test the work in this PR you will need to edit the `Hs search` view. Go to Structure > Views > Hs search > Edi
3. In the center column, find `PAGER`.
<img width="292" alt="pager" src="https://user-images.githubusercontent.com/12678977/78408453-2821a980-75d5-11ea-97a8-01501d9f86f7.png">

* Set `Use Pager` to `Full`
* Click on the text to the right of `Full` to update the `Use Pager` settings. In the screenshot above this is `Paged, 5 items`
* In the `Page: Pager options` module, change the `Pager ID` to `1`.
* You can also change the `Items per page` and `Number of pager links visible` as you please.
* Click Apply and Save.
4. Go to the home page and search for `seminar`.
5. Scroll to the bottom of the search results page and check out the pager. Click on various pagination links to verify it functions as expected.
6. After working on the pager I was curious if the updated `Pager ID` setting fixed the infinite scroll option and it does. To test infinite scroll return to the `Hs search` view and change `Use Pager` to `Infinite Scroll`.  Click Apply. The `Page: Pager options` module should open. Check the box to `Automatically Load Content` and click Apply. Save and test.
7. Note that even though the infinite scroll option  works, it is not recommended. Infinite scroll can result in accessibility issues. Sources: [Infinite Scroll & Role=Feed Accessibility Issues](https://www.deque.com/blog/infinite-scrolling-rolefeed-accessibility-issues/) and [Infinite Scrolling - Impact on Accessibility Series: #1 Common Issues](https://www.levelaccess.com/infinite-scrolling-impact-on-assistive-technologies-series-1/)

## Testing update
The work in this PR should also be reviewed using the `Custom Search` view. The styles written for the search and pagination elements should be consistent on both `HS Search` and `Custom Search` views. You can make the following changes to see the `Custom Search` view. Or you can search for a term and update the URL from http://economics.suhumsci.loc/search? to http://economics.suhumsci.loc/search-test?.
1. Go to `Structure` > `Block Layout`
2. Under the `HumSci Colorful` theme tab, find the `Search Block`
3. Click on the `Place block` button
4. Find `Exposed form: custm_custom_search-search_page` and `place block`
    Configure block:
    * `Display title` and `Override title` options are both unchecked
    * Set `Region` to `Search`
    * Save block
5. Disable `Exposed form: hs_search-search_page`
6. Scroll down and find the `Content` block
7. Repeat steps 3 through 5 above for the `Content` block with one additional step
    When configuring the block, go to the `Visibility` section and select `Pages`. Add `/search` to the `Pages` input. The `Show for listed pages` radio button should be selected. `Region` needs to be set to `Content`.
8. Save blocks


## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
